### PR TITLE
Playwright: migrate Domain Management: Add spec.

### PR DESCRIPTION
**Pull request by worldomonation**
#### Changes proposed in this Pull Request

This PR proposes to migrate the Domain Management: Add spec.

Key changes:
- minor restructuring of the original spec to reduce non-determinism.
- new components and pages to interact with underlying pages in the domain addition flow.

Background details:

The original Selenium version of the test had multiple weak points that introduced non-determinism. In the migration process, those elements have been refactored out while keeping the original purpose of the spec intact.

Otherwise, this is a straightforward, normal migration without any complications.

#### Testing instructions

- [x] Playwright mobile tests pass.
- [x] Playwright desktop tests pass.
- [x] Playwright Pre-Release tests pass.

Related to #

### CODE REVIEW

File: packages/calypso-e2e/src/lib/pages/domains-page.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import { Page } from 'playwright';
+
+const selectors = {
+	// Cart popover
+	cartButton: '.popover-cart button',
+	popOver: '.popover',
+	removeItemButton: 'button[aria-label="Remove item"]',
+	popOverCartPlaceholder: '.cart-item__loading-placeholder',
+
+	searchForDomainButton: `a:text-matches("(s|S)earch", "i")`,
```
Comment (WunderBart) [inline packages/calypso-e2e/src/lib/pages/domains-page.ts]: nit: the upper case should not be needed with the case-insensitive flag

```suggestion
	searchForDomainButton: `a:text-matches("search", "i")`,
```

...or the other way round:

```suggestion
	searchForDomainButton: `a:text-matches("(s|S)earch")`,
```

File: packages/calypso-e2e/src/lib/pages/domains-page.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import { Page } from 'playwright';
+
+const selectors = {
+	// Cart popover
+	cartButton: '.popover-cart button',
+	popOver: '.popover',
```
Comment (WunderBart) [inline packages/calypso-e2e/src/lib/pages/domains-page.ts]: nit: I believe it's a single word ([popover](https://en.wikipedia.org/wiki/Popover_(GUI)))

```suggestion
	popover: '.popover',
```

File: packages/calypso-e2e/src/lib/pages/domains-page.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+			const items = await this.page.$$( selectors.removeItemButton );
+			for await ( const item of items ) {
+				await item.click();
+				await this.page.waitForSelector( selectors.popOverCartPlaceholder, { state: 'hidden' } );
+			}
+		} else {
+			await this.page.click( `:nth-match( ${ selectors.removeItemButton }, ${ index })` );
+		}
+	}
+}
```
Comment (WunderBart) [inline packages/calypso-e2e/src/lib/pages/domains-page.ts]: IMO it would be easier to follow when separated into `removeCartItem` and `emptyCart`

**Review by WunderBart — APPROVED** [review]
Looks good! Just a few non-blockers 🚀

File: packages/calypso-e2e/src/lib/pages/domains-page.ts (hunk 1 above)
Comment (worldomonation) [reply packages/calypso-e2e/src/lib/pages/domains-page.ts]: Thanks!

File: packages/calypso-e2e/src/lib/pages/domains-page.ts (hunk 2 above)
Comment (worldomonation) [reply packages/calypso-e2e/src/lib/pages/domains-page.ts]: Got it!

File: packages/calypso-e2e/src/lib/pages/domains-page.ts (hunk 3 above)
Comment (worldomonation) [reply packages/calypso-e2e/src/lib/pages/domains-page.ts]: I'll revamp this and separate it.

**Pull request merged**